taxi/views.py

```diff
@@ -28,6 +28,21 @@ def index(request):
     return render(request, "taxi/index.html", context=context)
 
 
+@login_required
+def update_drivers(request, pk: int):
+    car = Car.objects.get(id=pk)
+    if request.user not in car.drivers.all():
+        car.drivers.add(request.user)
+        car.save()
+    else:
+        car.drivers.remove(request.user)
+        car.save()
```

Comment:
The `update_drivers` function directly manipulates the `drivers` ManyToMany field of the `Car` model. Consider using Django's `get_object_or_404` for retrieving the `Car` instance to handle cases where the car with the given `pk` does not exist. This will prevent the view from raising a `DoesNotExist` exception and instead return a 404 error page.